### DISCUSSION TITLE
Don't allow someone with commits in a PR to approve

### DISCUSSION THREAD

**Discussion by davich** [community/community#23134] [Pull Requests]
<p>We have our repo set up so all PRs must be approved by another engineer before they can be merged to main/master. You can not approve a PR that you have created. We rely on this to stop a single engineer getting code into production without someone else checking it.</p>
<p>But there is a (possible) security hole. A bad actor (eg disgruntled employee) could find a PR that is already open, revert the original changes, commit malicious code, then approve it and merge it.</p>
<p>Is there any way to stop someone from approving a PR if they have any commits in that PR?</p>
<p>Thanks</p>

**Comment by strafe** [comment]
<p>The issue is, the malicious person can commit code as someone else. You can try this yourself by changing your <code>user.name</code> and <code>user.email</code> config settings and making a commit. If they did this, any special mechanism you had setup to check for authors/committers would be foiled.</p>
<p>In theory if you had everyone GPG sign their commits and somehow checked for this too, it wouldn’t be an issue, but most people don’t GPG sign commits.</p>
<hr>
<p>What I’d suggest is requiring approval from 2 seperate people with write+ access to the repository if you have enough people to enforce this.</p>

**Comment by davich** [comment]
<p>Thanks for your reply! That’s interesting. I hadn’t considered that github has no way to verify whether the listed committer was actually the committer without people signing commits.</p>
<p>I wonder if you could say a user who has pushed to the PR can’t approve it. Surely Github knows who has pushed to a branch (you need to auth for that)</p>

**Comment by akloss-terion** [comment]
<p>I don’t think it’s true that GitHub can’t verify the committer… here’s a cut-n-paste from the UI:</p>
<blockquote>
<p>Test authored and <a href="https://github.com/cibotech/platform-definitions/commits?author=akloss-cibo" rel="noopener nofollow ugc">akloss-cibo</a> committed 2 minutes ago</p>
</blockquote>
<p>To me, this seems like a tragic gap in the pull request process.  Does anyone know something we don’t?</p>

**Comment by akloss-terion** [comment]
<p>GitLab has this feature:</p>
<blockquote>
<p><a href="https://docs.gitlab.com/ee/user/project/merge_requests/approvals/settings.html#prevent-approvals-by-users-who-add-commits" rel="noopener nofollow ugc">Prevent approvals by users who add commits</a> When enabled, users who have committed to a merge request cannot approve it.</p>
</blockquote>

**Comment by ettaboyle** [comment, marked as answer]
<p>Hi all! Tagging <a class="mention" href="/peckjon">@peckjon</a> on this :wave:t2:Jon just shared:</p>
<blockquote>
<p>I’ve built an Action to provide this; should work on <a href="http://GitHub.com">GitHub.com</a>, GHEC, and GHES: <a href="https://github.com/marketplace/actions/dismiss-code-reviews-from-collaborators" class="inline-onebox">Dismiss code reviews from collaborators · Actions · GitHub Marketplace · GitHub</a></p>
</blockquote>
<p><a class="mention" href="/davich">@davich</a> this post is originally from Jan 2021 so hoping you see this. Please let us know if this solves your issue.</p>
